File: src/trading/OptionExchange.ts
export interface OptionsExchange {
  name: string;

  placeOrder(params: {
    instrument_name: string;
    side: "buy" | "sell";
    price: number;
    quantity: number;
    order_type?: "limit" | "market";
    time_in_force?: "fill_or_kill";
  }): Promise<any>;
}

File: src_python/trading/option_exchange.py
from __future__ import annotations

from typing import Protocol


class OptionsExchange(Protocol):
    name: str

    async def place_order(
        self,
        *,
        instrument_name: str,
        side: str,
        price: float,
        quantity: float,
        underlying_price: float | None = None,
        order_type: str = "limit",
        time_in_force: str = "fill_or_kill",
    ) -> object:
        ...

